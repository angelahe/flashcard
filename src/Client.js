// interract with db back end

function createDeck() {
  return fetch('/api/deck', {
    method: 'POST',
  })
    .then(response => response.json())
    .then(json => json.id);
}

function createCard(L1word, L2word) {
  // let's try to send an api call with hardcoded values like in postman
  // return fetch('api/card?EN_word=test&ES_word=prueba', {
  // let's try to send the values passed in
  return fetch(`api/card?EN_word=${L1word}&ES_word=${L2word}`,
    {
    // return fetch('/api/card', {
      method: 'POST',
    })
    .then(response => response.json())
    .then(json => json.id);
}
const Client = { createDeck, createCard };
export default Client;
