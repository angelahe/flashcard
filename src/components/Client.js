// interract with db back end

function createDeck() {
  return fetch('/api/deck', {
    method: 'POST',
  })
    .then(response => response.json())
    .then(json => json.id);
}

function createCard(deckId, l1word, l2word) {
  // let's try to send an api call with hardcoded values like in postman
  // return fetch('api/card?EN_word=test&ES_word=prueba', {
  // let's try to send the values passed in
  // 127.0.0.1:5000/api/deck/68dfd811-9913-422f-8c50-037055314e13/card?EN_word=test&ES_word=prueba
  console.log(`args are deck:${deckId}, l1word: ${l1word}, l2word: ${l2word}!`);
  return fetch(`api/deck/${deckId}/card?EN_word=${l1word}&ES_word=${l2word}`,
  //  return fetch(`api/deck/card?EN_word=${l1word}&ES_word=${l2word}`,
    {
    // return fetch('/api/card', {
      method: 'POST',
    })
    .then(response => response.json())
    .then(json => json.id);
}
const Client = { createDeck, createCard };
export default Client;
