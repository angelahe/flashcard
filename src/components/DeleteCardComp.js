import React from 'react';
import PropTypes from 'prop-types';
import '../styles/flashcard.css';
import Client from './Client';
import done from '../img/buttons/done_FFFFFF.png';

class DeleteCardComp extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      errorMessage: '',
    };
  }

  handleConfirmDelete = () => {
    const { card, onCardDeleted } = this.props;
    Client.deleteCard(card)
      .then(response => this.setState({ errorMessage: response.message },
        () => onCardDeleted(card)));
  }

  render() {
    const { errorMessage } = this.state;
    return (
      <div className="DeleteCardComp">
        <br /> <br />
        <div className="LineContainer">
          <span className="DetailText">Confirm Delete? </span>
          <br /> <br />
        </div>
        <div className="LineContainer">
          <button className="AppBtn YesDelete" type="button" onClick={this.handleConfirmDelete}>
            <img className="btnImg" src={done} alt="Done" />
          </button>
        </div>
        { errorMessage
          ? <p>{errorMessage}</p>
          : null
        }
      </div>
    );
  }
}

DeleteCardComp.defaultProps = {
  onCardDeleted: () => { },
};

DeleteCardComp.propTypes = {
  card: PropTypes.string.isRequired,
  onCardDeleted: PropTypes.func,
};

export default DeleteCardComp;
