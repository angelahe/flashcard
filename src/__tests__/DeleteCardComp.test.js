import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import DeleteCardComp from '../components/DeleteCardComp';
import Client from '../components/Client';

const card = 'ABCD-1234';

it('renders without crashing', () => {
  shallow(<DeleteCardComp card={card} />);
});

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<DeleteCardComp card={card} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('snapshot of UI renders consistently', () => {
  const tree = renderer.create(<DeleteCardComp card={card} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('contains the container div EditDeckComp on shallow render', () => {
  const component = shallow(<DeleteCardComp card={card} />);
  expect(component.html()).toContain('DeleteCardComp');
});

it('should set the state when deleteCard completes', (done) => {
  const message = 'Successfully deleted';
  Client.deleteCard = jest.fn(() => Promise.resolve(message));

  const handleCardDeleted = jest.fn(() => {});
  const component = shallow(<DeleteCardComp card={card} onCardDeleted={handleCardDeleted} />);
  const button = component.find('.YesDelete').first();
  button.simulate('click');

  setTimeout(() => {
    // expect(component.instance().state.errorMessage).toBe(message);
    expect(handleCardDeleted.mock.calls.length).toBe(1);
    expect(handleCardDeleted.mock.calls[0][0]).toBe('ABCD-1234');
    done();
  });
});
