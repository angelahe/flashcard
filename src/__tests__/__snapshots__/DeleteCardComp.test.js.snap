// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot of UI renders consistently 1`] = `
<div
  className="DeleteCardComp"
>
  <br />
   
  <br />
  <div
    className="LineContainer"
  >
    <span
      className="DetailText"
    >
      Confirm Delete? 
    </span>
    <br />
     
    <br />
  </div>
  <div
    className="LineContainer"
  >
    <button
      className="AppBtn YesDelete"
      onClick={[Function]}
      type="button"
    >
      <img
        alt="Done"
        className="btnImg"
        src="done_FFFFFF.png"
      />
    </button>
  </div>
</div>
`;
